src/Collection.php

```diff
@@ -271,19 +283,11 @@ public function bulkWrite(array $operations, array $options = [])
      */
     public function count($filter = [], array $options = [])
     {
-        if (! isset($options['readPreference']) && ! is_in_transaction($options)) {
-            $options['readPreference'] = $this->readPreference;
-        }
-
-        $server = select_server($this->manager, $options);
-
-        if (! isset($options['readConcern']) && ! is_in_transaction($options)) {
-            $options['readConcern'] = $this->readConcern;
-        }
+        $options = $this->inheritReadOptions($options);
 
         $operation = new Count($this->databaseName, $this->collectionName, $filter, $options);
 
-        return $operation->execute($server);
+        return $operation->execute(select_server($this->manager, $options));
```

Comment:
This is a subtle behavioral change because we're now deferring server selection until _after_ constructing the Count operation, but I'll note that other helper methods vary in this regard.

For examples, helpers like `aggregate()` must select servers in advance to handle option inheritance.

I think it does make sense to defer server selection as long as possible, since it makes it easier to catch other client-side validation errors sooner.